CRM/Core/Permission.php

```diff
@@ -1486,8 +1486,8 @@ public static function getEntityActionPermissions() {
 
     $permissions['message_template'] = array(
       'get' => array('access CiviCRM'),
-      'create' => array('edit message templates', 'edit user-driven message templates', 'edit system workflow message templates'),
-      'update' => array('edit message templates', 'edit user-driven message templates', 'edit system workflow message templates'),
+      'create' => array('edit message templates'),
```

Comment:
This would have 'worked' had the array structure been different - see the double array in this place. However, I can see that your version precise than this

```
      'default' => array(
        // At minimum the user needs one of the following. Finer-grained access is controlled by CRM_Case_BAO_Case::addSelectWhereClause
        array('access my cases and activities', 'access all cases and activities'),
      ),
    );
```